ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -489,58 +869,82 @@ def _useRealization(self, info, rlz):
           g.data[index, constIndex] = self._handleExplicitConstraints(newOpt, constraint)
         else:
           g.data[index, constIndex] = self._handleImplicitConstraints(newOpt, opt, constraint)
+
     offSpringFitness = self._fitnessInstance(rlz,
                                              objVar=self._objectiveVar,
                                              a=self._objCoeff,
                                              b=self._penaltyCoeff,
-                                             penalty=None,
                                              constraintFunction=g,
-                                             type=self._minMax)
+                                             constraintNum = self._numOfConst,
+                                             type = self._minMax)
+    return traj, g, objectiveVal, offSprings, offSpringFitness
 
-    self._collectOptPoint(rlz, offSpringFitness, objectiveVal,g)
-    self._resolveNewGeneration(traj, rlz, objectiveVal, offSpringFitness, g, info)
+  #########################################################################################################
+  # Run Methods                                                                                           #
+  #########################################################################################################
+
+  #########################################################################################################
+  # Developer note:
+  # Each algorithm step is indicated by a number followed by the generation number
+  # e.g., '0 @ n-1' refers to step 0 for generation n-1 (i.e., previous generation)
+  # for more details refer to GRP-Raven-development/Disceret_opt channel on MS Teams.
+  #########################################################################################################
 
+  def _useRealization(self, info, rlz):
+    """
+      Used to feedback the collected runs into actionable items within the sampler.
+      This is called by localFinalizeActualSampling, and hence should contain the main skeleton.
+      @ In, info, dict, identifying information about the realization
+      @ In, rlz, xr.Dataset, new batched realizations
+      @ Out, None
+    """
+
+    info['step'] = self.counter
+
+    # 0 @ n-1: Survivor Selection from previous iteration (children+parents merging from previous generation)
+    # 0.1 @ n-1: fitnessCalculation(rlz): Perform fitness calculation for newly obtained children (rlz)
+
+    objInd = int(len(self._objectiveVar)>1) + 1 #if len(self._objectiveVar) == 1 else 2
+    constraintFuncs: dict = {1: GeneticAlgorithm.singleObjectiveConstraintHandling, 2: GeneticAlgorithm.multiObjectiveConstraintHandling}
+    const = constraintFuncs.get(objInd, GeneticAlgorithm.singleObjectiveConstraintHandling)
+    traj, g, objectiveVal, offSprings, offSpringFitness = const(self, info, rlz)
```

Comment:
Similarly to before, I suggest to create a self variable to indicate if the analysis is multi-objective or not.